### DISCUSSION TITLE
Jump in & Jump out should not unfold nodes

### DISCUSSION THREAD

**Discussion by sifran-github** [freeplane/freeplane#928] [General]
following @dpolivaev advice, I created this thread to ask our community about what do you prefer.

**_THE ISSUE_**
Today, when you jump in & jump out of a folded node, the node gets unfolded.

**_MY OPINION_**
I think that this is a bug and it should not fold the node after you jump out.

**_REASONS_**
1. These are two separate attributes of a node (jumping & folding).
2. I see a very common usage of Jump in & Jump in which the user is having a large map with some nodes that are folded (they contain a lot of children and information). The user wants to keep them folded and look at "the big picture". When the user wants to edit the content, he can just:
a. click on a folded node, jump in...
b. edit the content.
c. jump out to the main view of the mindmap.

What do you think?

**Comment by euu2021** [comment]
Yes, I agree.  It was in my tasklist to make this suggestion, for the exact reason that you explained.

**Comment by macmarrum** [comment]
I support the idea to segregate jump in/out and (un)folding.

**Comment by sifran-github** [comment]
I see that in freeplane_bin-1.11.1-pre12 it has been changed, but it now behaves differently than what i expected, and actually even worse than before...
If you read in my original post, you'll see that the whole reason (number 2) for this post was to quickly jump-edit-exit folded nodes.

but the current fix is making it even harder. the current fix keeps the node folded ALSO IN THE JUMP-IN EDITING SCREEN.
So currently in order to do what i want, i have to:
1. jump in
2. unfold the node
3. edit the content
4. fold the node
5. jump out

Steps 2 (manually Unfolding) and 4 (manually folding) are the problem. They are redundant and i'm trying to avoid these steps, BUT keeping the node folded. Heres how:

My idea was that if you jump into a node, it should behave like a root node. root nodes cannot be folded and they stay open all the time. so even if the node status is folded on the main map view, IT WILL BE SHOWN AS IF IT IS A TEMP-ROOT, UNFOLDED when you jump into it, BUT it will remember the status of the node for the main map view (after you jump out). Meaning, that it will behave like root node, always showing the children of the temp-root and the user will not be given an option to fold or unfold the temp-root.

Then, after editing the content in the jump-in view and jumping out, the node should be shown again like a regular node and not a temp-root node, which means that if the node's original state was folded, it should stay folded.

This is what i intended...

**Reply by dpolivaev** [reply]
I changed it again in Preview 1.11.1-14, https://sourceforge.net/projects/freeplane/files/freeplane%20preview/

**Comment by sifran-github** [comment]
## It is working well
Tested it in Preview 1.11.1-14, it seems to work as expected. thanks!

I'm actually impressed how well it works now, and how nice it is to have folded nodes with a lot more information inside them, and then just quickly jump in & out. It even open them in their own layout configuration, so i can configure which layout i want to see once i jump in a specific node. very useful!

I also like that you can have multiple levels of jump ins: you can jump in to a folded node, then jump into his child node which is also folded... and then jump out twice. works perfectly.

## Suggestions to simplify the usage jump in & jump out

Anyway I can configure these suggestions on my Freeplane Installation (Its really amazing that you can configure almost everything in Freeplane!), but I suggest to make these default:

1.  jump in & jump out keyboard shortcuts: my windows doesn't let me use the ctrl+esc because it configured by windows to open the start menu, so only shift+esc works for me. Maybe jump in and out by ctrl+> and ctrl+< instead of esc key? (right now they are used by notes, which i think can have the keyboard shortcut ctrl+shift+N instead)
2.  jump in & jump out toolbar icons: add them to the toolbar. (Anyway I will do it like I did with text alignment: https://github.com/freeplane/freeplane/issues/918)

**Reply by sifran-github** [reply]
I just noticed how ESC is being handled, and its very good: pressing ESC goes to the temp root. pressing again, jumping out and still selecting the root. pressing again, goes to the root. its really good.

I guess that with the current configuration, even if windows takes over CTRL+ESC, I can still use just ESC (or clicking ESC twice) in order to jump out.
